Added Image Segmentation in CV Basics

PiyushPatle26 opened the pull request:
In the new version of the Pixels_seminar project, I incorporated an image segmentation functionality. Using this functionality, the system can segment an image into regions and map each region to different objects or regions of the scene. With the pixel-by-pixel labeling of pixels based on properties like color, intensity, or texture, the system can detect and analyze different image regions better.

AryanNanda17 commented on 4_cv_basics/9_image_segmentation/README.md:
```diff
@@ -0,0 +1,168 @@
+- K-means clustering can be implemented using `cv::kmeans()`
+
+Each approach has its strengths and weaknesses, making them suitable for different applications and image types. By understanding the theoretical foundations and practical implementations of these techniques, you'll be well-equipped to tackle a wide range of image segmentation challenges.
+## How to Use This Project
```
Use Running Instructions

AryanNanda17 commented on 4_cv_basics/9_image_segmentation/src/main.cpp:
Move main.cpp under root directory.

AryanNanda17 replied:
Accordingly change the include paths

AryanNanda17 commented on install.sh:
```diff
@@ -57,17 +57,14 @@ if ! command_exists git; then
 fi
 
-if [ ! -d "Pixels_Seminar" ]; then
+if [[ ! -d Pixels_Seminar && "$(basename "$PWD")" != "Pixels_Seminar" ]]; then
```
Please remove this

AryanNanda17 commented on install.sh:
No need to make changes to this file

AryanNanda17 commented on 4_cv_basics/9_image_segmentation/README.md:
```diff
@@ -0,0 +1,168 @@
+- Draw a rectangle around the object using the mouse.
+- The program applies GrabCut to extract the foreground object.
+- The result is displayed with a clean white background.
+
```
Add images here after every step

AryanNanda17 commented on 4_cv_basics/9_image_segmentation/README.md:
```diff
@@ -0,0 +1,168 @@
+  1, & \text{if } I(x,y) \geq T \\
+  0, & \text{if } I(x,y) < T
+  \end{cases} $$
+- Color: Using color similarity metrics
```
Correct the latex sytax

AryanNanda17 commented on 4_cv_basics/9_image_segmentation/README.md:
```diff
@@ -0,0 +1,168 @@
+  $$ S(x,y) = \begin{cases} 
+  1, & \text{if } I(x,y) \geq T \\
+  0, & \text{if } I(x,y) < T
+  \end{cases} $$
```
correct the sytax

AryanNanda17 reviewed (changes requested):
You did an amazing job @PiyushPatle26. Structure, code quality, format everything looks good to me. Please do the requested changes and this is good to go.

AryanNanda17 commented:
@Herculoxz
Please review the pr once.

Herculoxz commented:
apart from the suggested changes , LGTM

AryanNanda17 commented on 4_cv_basics/9_image_segmentation/Makefile:
```diff
@@ -0,0 +1,29 @@
+	$(error "link is not set")
+endif
+	@echo "Building..."
+	@$(CC) main.cpp $(link) -o $(PROJECT) $(CFLAGS) $(LIBS)
```
why did you remove src? don't remove. run it like make src=main.cpp link=""

AryanNanda17 commented on 4_cv_basics/9_image_segmentation/README.md:
```diff
@@ -0,0 +1,204 @@
+4. [How GraphCut Works](#how-graphcut-works)
+5. [Additional Information](#additional-information)
+6. [Further Learning Resources](#further-learning-resources)
+7. [How to Use This Project](#how-to-use-this-project)
```
Running Instructions!

AryanNanda17 commented on 4_cv_basics/9_image_segmentation/README.md:
````diff
@@ -0,0 +1,204 @@
+   ```bash
+   git clone https://github.com/SRA-VJTI/Pixels_Seminar.git
+   cd 4_cv_basics/9_image_segmentation
+   ```
````
no need to do this. See running instructions of blob_detection for example. please maintain consistency.

AryanNanda17 commented on 4_cv_basics/9_image_segmentation/README.md:
````diff
@@ -0,0 +1,204 @@
+
+2. **Build the Program**
+   ```bash
+   make build
````
src, link?

AryanNanda17 commented on 4_cv_basics/9_image_segmentation/README.md:
```diff
@@ -0,0 +1,204 @@
+### Prerequisites
+- OpenCV 4.x installed
+- C++ compiler (g++)
+- Make build system
```
Please remove this and How to use this project. This directory is part of the Pixels_Seminar repo and this is already there in the main readme. 

Only keep running instructions.

AryanNanda17 reviewed (changes requested):
Please do the required changes

AryanNanda17 commented:
@SuperChamp234, what do you think of this example?

SuperChamp234 commented:
This example is amazing! is it possible for FYs to do the implementation of GrabCut as part of an assignment?

Herculoxz commented on 4_cv_basics/9_image_segmentation/README.md:
```diff
@@ -0,0 +1,204 @@
+1. **Watershed Algorithm**: A region-based approach that treats an image as a topographic surface
+2. **Mean Shift Segmentation**: A feature-space analysis technique for locating maxima of a density function
+3. **K-means Clustering**: A simple but effective method for partitioning pixels into k clusters
+4. **Deep Learning Approaches**: Neural network architectures like U-Net or Mask R-CNN that have revolutionized semantic segmentation
```
Include hyperlinks to the official OpenCV documentation for each of these concepts.

PiyushPatle26 commented:
done with the suggested changes

AryanNanda17 commented:
> This example is amazing! is it possible for FYs to do the implementation of GrabCut as part of an assignment?

I think it would be difficult for them to do it at this point.

AryanNanda17 commented:
@PiyushPatle26, whenever you make commits, please write messages that clearly describe what you have changed or implemented. It's very hard to figure it out otherwise.

PiyushPatle26 commented:
> @PiyushPatle26, whenever you make commits, please write messages that clearly describe what you have changed or implemented. It's very hard to figure it out otherwise.

Okay i'll keep this in mind from next time.
i just changed the makefile so that it make build commands needs src and link 
and changed the readme as the requested changes , added hyperlinks in readme , removed prerequisites and also changed format.

PiyushPatle26 commented:
> @PiyushPatle26, whenever you make commits, please write messages that clearly describe what you have changed or implemented. It's very hard to figure it out otherwise.

Okay i'll keep this in mind from next time.
i just changed the makefile so that it make build commands needs src and link 
and changed the readme as the requested changes , added hyperlinks in readme , removed preqrequisites and also changed format.

AryanNanda17 reviewed (approved):
/lgtm
